src/comm.jl

```diff
@@ -1,264 +1,1531 @@
+               for field in fields
+                  for i in 0:(dx-1)
+                     domain.commDataSend[offset+i] = field[i]
+                  end
+                  offset += dx
+               end
+               idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + 1
+               src = view(domain.commDataSend, idx:(idx+(xferFields * dx)))
+               otherRank = myRank - domain.tp^2 - domain.tp
+               req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+               domain.sendRequest[pmsg+emsg+1] = req
+               emsg += 1
+            end
+
+         if colMin && planeMin
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm
+            for field in fields
+               for i in 0:(dy-1)
+                  domain.commDataSend[offset+i] = field[i*dx]
+               end
+               offset += dy
+            end
+            idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + 1
+            src = view(domain.commDataSend, idx:(idx+(xferFields * dy)))
+            otherRank = myRank - domain.tp^2 - 1
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+1] = req
+            emsg += 1
+         end
+
+         if rowMax && colMax && doSend
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm
+            srcOffset = dx*dy - 1
+            for field in fields
+               for i in 0:(dz-1)
+                  domain.commDataSend[offset+i] = field[srcOffset+i*dx*dy]
+               end
+               offset += dz
+            end
+            idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + 1
+            src = view(domain.commDataSend, idx:(idx+(xferFields * dz)))
+            otherRank = myRank + domain.tp + 1
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+1] = req
+            emsg += 1
+         end
+
+         if rowMax && planeMax && doSend
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm
+            srcOffset = dx*(dy-1) + dx*dy*(dz-1)
+            for field in fields
+               for i in 0:(dx-1)
+                  domain.commDataSend[offset+i] = field[srcOffset+i]
+               end
+               offset += dx
+            end
+            idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + 1
+            src = view(domain.commDataSend, idx:(idx+(xferFields * dx)))
+            otherRank = myRank + domain.tp^2 + domain.tp
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+1] = req
+            emsg += 1
+         end
+
+         if colMax && planeMax && doSend
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm
+            srcOffset = dx*dy*(dz-1) + dx
+            for field in fields
+               for i in 0:(dy-1)
+                  domain.commDataSend[offset+i] = field[srcOffset+i*dx]
```

Comment:
commDataSend (not field) require +1, I would however move the -1 (implicitly deleted) back into srcOffset and add an explicit +1 for consistency sake.

Reply:
Not sure I follow, why does field not require `+1`?